operator-framework-core/src/main/java/io/javaoperatorsdk/operator/processing/dependent/workflow/WorkflowCleanupResult.java

```diff
@@ -1,30 +1,22 @@
 package io.javaoperatorsdk.operator.processing.dependent.workflow;
 
 import java.util.List;
-import java.util.Map;
 
 import io.javaoperatorsdk.operator.api.reconciler.dependent.DependentResource;
 
 @SuppressWarnings("rawtypes")
-public class WorkflowCleanupResult extends WorkflowResult {
-  private Boolean allPostConditionsMet;
+public interface WorkflowCleanupResult extends WorkflowResult {
```

Comment:
Can this kind of change go in not a major release? This is a public class so it will break user code if someone extended it originally and now it's an interface.

Reply:
Theoretically, yes. Practically, these classes are not meant to be extended by the user, they're objects that the SDK returns as part of its processing. The only case where I could potentially imagine a user extending these classes would be in tests but even so, that would be quite a stretch, in my opinion…

Reply:
This targets next, what will be a major release.

Reply:
Also, yes! Forgot about that! 🤦😅